src/components/Product/ProductsFilterLabels/ProductsFilterLabels.tsx

```diff
@@ -1,24 +1,66 @@
 'use client'
+import { useEffect, useState } from 'react'
 import Image from 'next/image'
-import { IProductFilterLabel } from '@/types/IProductFilterLabel'
 import Button from '@/components/UI/Buttons/Button/Button'
 import { twMerge } from 'tailwind-merge'
+import {
+  defaultProductsFilters,
+  useProductFiltersStore,
+} from '@/store/productFiltersStore'
+
+type FilterType = 'seller' | 'brand'
+type FilterLabelItemType = {
+  value: string
+  filterType: FilterType
+}
+
+const buildFilterLabelItems = (
+  brands: string[] = [],
+  sellers: string[] = [],
+): FilterLabelItemType[] => {
+  const brandItems = brands.map(
+    (brand) => ({ value: brand, filterType: 'brand' }) as FilterLabelItemType,
+  )
+  const sellerItems = sellers.map(
+    (brand) => ({ value: brand, filterType: 'seller' }) as FilterLabelItemType,
+  )
+
+  return [...brandItems, ...sellerItems]
+}
 
 interface IProductsFilterLabels {
-  filterLabels: IProductFilterLabel[]
-  handleFilterByDefault: () => void
-  handleFilterLabelClick: (label: string, id: string) => void
   className?: string
 }
 
 const ProductsFilterLabels = ({
-  filterLabels,
-  handleFilterLabelClick,
-  handleFilterByDefault,
-  className
+  className,
 }: Readonly<IProductsFilterLabels>) => {
+  const {
+    selectedBrandOptions,
+    selectedSellerOptions,
+    removeSellerOption,
+    removeBrandOption,
+    updateProductFiltersStore,
+  } = useProductFiltersStore()
+
+  const [filterLabels, setFilterLabels] = useState(() =>
+    buildFilterLabelItems(selectedBrandOptions, selectedSellerOptions),
+  )
+
+  useEffect(() => {
+    setFilterLabels(
+      buildFilterLabelItems(selectedBrandOptions, selectedSellerOptions),
+    )
+  }, [selectedSellerOptions, selectedBrandOptions, setFilterLabels])
```

Comment:
we can delete setFilterLabels from dep array, but there is no difference in work